code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -76,14 +76,16 @@
 	if(istype(H))
 		var/obj/item/card/id/I = H.wear_id
 		if(!istype(I) || !check_access(I))
-			H.visible_message(SPAN_NOTICE("[src] beeeps as [H] picks it up"), SPAN_DANGER("WARNING: Unauthorized user detected. Denying access..."))
-			H.KnockDown(20)
+			var/datum/effect_system/spark_spread/s = new
+			s.set_up(5, 1, src.loc)
+			H.visible_message(SPAN_NOTICE("[src] beeps as [H] picks it up"), SPAN_DANGER("WARNING: Unauthorized user detected. Denying access..."))
 			H.visible_message(SPAN_WARNING("[src] beeps and sends a shock through [H]'s body!"))
+			H.emote("pain")
+			s.start()
 			deductcharge(hitcost)
```

Comment:
Needs a check earlier for if it has sufficient charge or this might put the cell into the negative

Reply:
Fixed. If the stun baton doesn't have power, it doesn't auth check and can therefore be picked up.